### DISCUSSION TITLE
Decompression speed解压速度

### DISCUSSION THREAD

**Discussion by nenge123** [gildas-lormeau/zip.js#353] [Q&A] [answered]
我使用.“WinRAR”.
分别使用标准和储存两个方式压缩,254MB的"Dragon quest monsters 1&2.img".

文件情况
----
- 标准 122MB
- 储存 265MB

解压方式
----
- BlobReader(File)
  - 标准:无卡顿，有进度，耗时2:50分钟。
  - 存储:无卡顿，有进度，耗时30秒。

-  Uint8ArrayReader(new Uint8Array)
   - 标准:无卡顿，有进度，耗时2:30分钟。
   - 储存:界面差点无响应，卡顿。并且无解压进度。但是卡顿结束会显示瞬间解压完毕。

问题
----
- “Uint8Array”过大会严重卡顿，Edge会提示‘无响应，是否关闭，或者等待’

- 你所说的几秒是什么模式解压！？

**Comment by nenge123** [comment]
if using  ArrayBuffer.slice

class Uint8ArrayReader{
  buff = buf.buffer; ?
}

统一化所有截取值都是"ArrayBuffer"

**Comment by 0f-0b** [comment, marked as answer]
使用了 `Uint8ArrayWriter` 吗？如果是，那么换成下面的实现或者 `BlobWriter` 之后，情况有所改善吗？

```js
class Uint8ArrayWriter extends Writer {
  chunks = [];

  writeUint8Array(array) {
    super.writeUint8Array(array);
    this.chunks.push(array);
  }

  getData() {
    const { chunks } = this;
    let size = 0;
    for (const chunk of chunks) {
      size += chunk.length;
    }
    const array = new Uint8Array(size);
    let offset = 0;
    for (const chunk of chunks) {
      array.set(chunk, offset);
      offset += chunk.length;
    }
    return array;
  }
}
```

**Reply by nenge123** [reply]
oh my god! that fastest!
这是什么原因？？

如果uint8array那么慢，为什么不统一使用BlobWrite?

at last return new Uint8array (await Blob.arrayBuffer())

**Reply by nenge123** [reply]
参考这个代码我找到了修复方案,差不多一样快!

**Comment by nenge123** [comment]
@gildas-lormeau 

```javascript
	class Uint8ArrayWriter extends Writer {

		constructor() {
			super();
			this.array = [];
		}

		writeUint8Array(array) {
			super.writeUint8Array(array);this.array.push(e);
		}

		async getData() {
			return new Uint8Array(await new Blob(this.array).arrayBuffer())
		}
	}
```

**Reply by nenge123** [reply]
fast same as BlobWriter

**Comment by gildas-lormeau** [comment]
Thank you, it's fixed in the version 2.6.14 I just published.
